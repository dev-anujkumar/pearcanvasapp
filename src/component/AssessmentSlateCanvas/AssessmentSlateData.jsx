// IMPORT - Plugins //
import React, { Component } from 'react';

// IMPORT - Assets //
import config from '../../config/config';
import './../../styles/AssessmentSlateCanvas/AssessmentSlateCanvas.css';
import { showTocBlocker, hideTocBlocker, disableHeader } from '../../js/toggleLoader';
import { assessmentUsageType, assessmentType, FULL_ASSESSMENT_PUF, LEARNING_APP_TYPE, LEARNOSITY, LEARNING_TEMPLATE, PUF, FULL_ASSESSMENT_CITE, FULL_ASSESSMENT_TDX  } from './AssessmentSlateConstants.js';
// import RootElmComponent from './elm/RootElmComponent.jsx';
import RootElmSingleAssessment from '../AssessmentSlateCanvas/elm/RootElmSingleComponent.jsx';
import RootCiteTdxComponent from './assessmentCiteTdx/RootCiteTdxComponent.jsx';
import LearningTool from './learningTool/learningTool.jsx';
import { sendDataToIframe, hasReviewerRole, setAssessmentUsageType } from '../../constants/utility.js';
import { ShowLoader } from '../../constants/IFrameMessageTypes.js';
import  {setCurrentCiteTdx,assessmentSorting}  from '../AssessmentSlateCanvas/assessmentCiteTdx/Actions/CiteTdxActions';
import { connect } from 'react-redux';

 class AssessmentSlateData extends Component {
    constructor(props) {
        super(props);
        this.state = {
            activeAssessmentType: this.props.model && this.props.model.elementdata && this.props.model.elementdata.assessmentformat && this.props.model.elementdata.assessmentformat!== 'fpo'? this.props.model.elementdata.assessmentformat : 'Select',
            activeAssessmentUsageType: setAssessmentUsageType(this.props),//this.props.model && this.props.model.elementdata && this.props.model.elementdata.usagetype ? this.props.model.elementdata.usagetype : "Quiz",
            showElmComponent: false,
            changeLearningData: false,
            learningToolStatus: false,
            showCiteTdxComponent:false,
            parentPageNo:1,
            isReset: false,
            searchTitle : '',
            filterUUID : ''
        }
        this.usageTypeDropdownRef = React.createRef();
        this.typeDropdownRef = React.createRef();
        this.usageTypeRef = React.createRef();
        this.typeRef = React.createRef();
    }

    resetPage = (isReset, isSearch=false) => {
        this.setState({isReset})
        if(isReset && isSearch){
            this.setState({parentPageNo:1})
        } else if (isReset){
            this.setState({parentPageNo:1})
            this.setState({searchTitle:'', filterUUID:''})
        }
    }

    AssessmentSearchTitle = (searchTitle, filterUUID) => {
        this.setState({searchTitle, filterUUID},()=>{
            console.log("SetSate for filter", searchTitle + filterUUID)
        });
    }
    
    componentWillReceiveProps(nextProps){
        
     if(this.props.getAssessmentDataPopup !== nextProps.getAssessmentDataPopup){
         this.sendDataAssessment(nextProps);
     }
    }

    sendDataAssessment(nextProps){
        if(config.parentEntityUrn !== "Front Matter" && config.parentEntityUrn !== "Back Matter" && config.slateType =="assessment"){
            let apiKeys = [config.ASSET_POPOVER_ENDPOINT,config.STRUCTURE_APIKEY,config.PRODUCTAPI_ENDPOINT];
            let assessmentId= nextProps.model.elementdata.assessmentid.length>0 ? nextProps.model.elementdata.assessmentid: '' ;
            if(assessmentId!=""){
                sendDataToIframe({ 'type': 'getAssessmentLO', 'message': { projectURN: config.projectUrn, assessmentId, apiKeys} });
            }
            else{
               //set tag to grey here 
               let newMessage = {assessmentResponseMsg:false};
               this.props.isLOExist(newMessage);
            }
            
        }
    }

    componentDidMount() {
        let newMessage = {assessmentResponseMsg:false};
        this.props.isLOExist(newMessage);
        if (this.props.model && this.props.model.elementdata && this.props.model.elementdata.assessmentid) {
            this.sendDataAssessment(this.props);
            this.setState({
                activeAssessmentType: this.props.model && this.props.model.elementdata && this.props.model.elementdata.assessmentformat && this.props.model.elementdata.assessmentformat!== 'fpo'? this.props.model.elementdata.assessmentformat : 'Select',
            })
        }
    }

    /*** @description - This function is to link learning app*/
    linkLearningApp = (selectedLearningType) =>{
        sendDataToIframe({'type': ShowLoader,'message': { status: true }});
        this.props.updateAssessment(selectedLearningType.learningtemplateUrn,"",selectedLearningType.label.en,LEARNING_TEMPLATE,this.state.activeAssessmentUsageType,'insert',selectedLearningType.learningsystem,selectedLearningType.templateid,selectedLearningType.type);
        this.props.closeLtAction();
    }

    /*** @description - This function is to close ELM PopUp
    */
    closeElmWindow = () => {
        this.setState({
            showElmComponent: false
        });
        hideTocBlocker(false);
        disableHeader(false);
        this.props.showBlocker(false);
    }
    /*** @description - This function is to close CITE/TDX PopUp
    */
   closeWindowAssessment = () => {
        this.props.setCurrentCiteTdx({});
        this.setState({
            showCiteTdxComponent: false
        });
        hideTocBlocker(false);
        disableHeader(false);
        this.props.showBlocker(false);
    }
    
    /*** @description - This function is to change the lerning system
    */
    changeLearningApp() {
        showTocBlocker();
        disableHeader(true);
        this.props.showBlocker(true);
        //Call this function to set value of "toggleLT" for conditional based rendering of Learning App Component//
        this.props.openLTFunction(); 
        this.props.openLtAction();  
        this.setState({
            activeAssessmentType: LEARNING_TEMPLATE,
            changeLearningData: true,
            learningToolStatus: true
        });         
    }

    /*** @description - This function is to handle change in assessment/LT-LA
     * @param e- event triggered
    */
    changeAssessment = (e) => {
        if(hasReviewerRole()){
            return true
        }
        let assessmentFormat = this.state.activeAssessmentType;
        if (assessmentFormat === FULL_ASSESSMENT_PUF || assessmentFormat == PUF ) {
            this.setState({
                activeAssessmentType: FULL_ASSESSMENT_PUF,
                showElmComponent: true,
            }, () => {
                    this.mainAddAssessment(e, FULL_ASSESSMENT_PUF);
                })
        } else if (assessmentFormat === LEARNOSITY || assessmentFormat == "learnosity") {
            this.setState({
                activeAssessmentType: LEARNOSITY,
                showElmComponent: true,
            }, () => {
                    this.mainAddAssessment(e, LEARNOSITY);
            })
        // } else if (assessmentFormat === LEARNING_TEMPLATE) {
        } else 
        if (assessmentFormat === LEARNING_TEMPLATE || assessmentFormat === LEARNING_APP_TYPE) {   //PCAT-6773 fixed
            this.mainAddAssessment(e, LEARNING_TEMPLATE);
            } 
            else if(assessmentFormat === "cite" || assessmentFormat === "tdx" || assessmentFormat === FULL_ASSESSMENT_TDX || assessmentFormat ===FULL_ASSESSMENT_CITE){
                this.props.assessmentSorting("","");
                this.setState({
                    activeAssessmentType: (assessmentFormat==="cite" || assessmentFormat === FULL_ASSESSMENT_CITE)?FULL_ASSESSMENT_CITE:FULL_ASSESSMENT_TDX,
                    showCiteTdxComponent: true
                }, () => {
                    this.mainAddAssessment(e, (assessmentFormat==="cite" || assessmentFormat === FULL_ASSESSMENT_CITE)?FULL_ASSESSMENT_CITE:FULL_ASSESSMENT_TDX);
            })
            }
            else {
            this.addC2MediaAssessment(this.state.activeAssessmentType);
        }
    }

    /*** @description - This is the function to add C2-Media to Assessment Slate 
     * @param activeAssessmentType - assessment -type 
    */
    addC2MediaAssessment = (activeAssessmentType) => {
        this.props.toggleAssessmentPopup('',true);
        this.props.selectAssessmentType(activeAssessmentType);

    }

    /* 
    * @description - This function is to close the LT/LA popup 
    */
    closePopUp = () => {
        this.setState({
            changeLearningData: false,
           
        }, () => {
             disableHeader(false);
             hideTocBlocker(false);
             this.props.showBlocker(false);
             setTimeout(() => {
                this.setState({
                    learningToolStatus: false
                })
            }, 5000)
        })
    }
    closelearningPopup = () => {
        disableHeader(false);
        hideTocBlocker(false);
        this.props.showBlocker(false);
        this.setState({
            activeAssessmentType: LEARNING_APP_TYPE,
            changeLearningData: false,
            learningToolStatus: false
        })
    }
    
    /*** @description - This is the function to add C2-Media to Assessment Slate 
    * @param pufObj - The object contains data about PUF Assessment 
    */
    addPufAssessment = (pufObj) => {
        this.props.addPufAssessment(pufObj , this.state.activeAssessmentType);
    }
    addCiteTdxAssessment =(citeTdxObj) =>{
        this.props.addCiteTdxAssessment(citeTdxObj,this.state.activeAssessmentType);
    }

    /*** @description - This is the root function to add Assessment 
     * @param activeAssessmentType - assessment -type 
    */
    mainAddAssessment = (e, activeAssessmentType) => {
        if(this.props.permissions && this.props.permissions.includes('quad_create_edit_ia')){
        switch (activeAssessmentType) {
            case LEARNING_TEMPLATE:
            case LEARNING_APP_TYPE: 
                sendDataToIframe({ 'type': 'hideToc', 'message': {} });              
                this.changeLearningApp()
                break;
            case FULL_ASSESSMENT_PUF:
            case LEARNOSITY:
                this.setState({
                    showElmComponent: true
                })
                sendDataToIframe({ 'type': 'hideToc', 'message': {} });
                showTocBlocker(true);    
                disableHeader(true);
                this.props.showBlocker(true);             
                break;
            case FULL_ASSESSMENT_CITE:
            case FULL_ASSESSMENT_TDX:
                    sendDataToIframe({ 'type': 'hideToc', 'message': {} });
                    this.props.assessmentSorting("","");
                    this.setState({
                        showCiteTdxComponent: true
                    })
                showTocBlocker();
                disableHeader(true);
                this.props.showBlocker(true);
                break;
            default:
                this.addC2MediaAssessment(activeAssessmentType)
                break;

        }
    }
    }

    /*** @description - This function is to toggle the Assessment Type PopUp*/
    toggleAssessmentTypeDropdown = () => {
        if (this.typeRef.current.classList.contains('notselect')) {
            this.typeRef.current.classList.remove('notselect');
            this.typeDropdownRef.current.classList.remove('notselect')
        } else {
            this.typeRef.current.classList.add('notselect');
            this.typeDropdownRef.current.classList.add('notselect')
        }
    }

    /*** @description - This function is to handle the Assessment type change
     * @param type - the type of assessment selected from the dropdown
     * @param e - event triggered 
    */
    handleAssessmentTypeChange = (type, e) => {
        this.setState({
            activeAssessmentType: type,
        });
        this.typeDropdownRef.current.classList.add('notselect')
        this.typeRef.current.classList.add('notselect')
    }

    /*** @description - This function is to select the Assessment type from dropdown*/
    selectAssessmentType = () => {
        if(hasReviewerRole()){
            return true
        }
        let assessmentTypeValue;
        if (assessmentType.length > 0) {
            assessmentTypeValue = assessmentType.map((type, i) =>
                <li key={i} className="slate_assessment_dropdown_name" onClick={(e) => this.handleAssessmentTypeChange(type, e)}>{type}</li>
            )
        }
        return assessmentTypeValue
    }

    /*** @description - This function is to toggle the Assessment Usage-Type PopUp*/
    toggleUsageTypeDropdown = () => {
        if (this.usageTypeDropdownRef.current.classList.contains('notselect')) {
            this.usageTypeDropdownRef.current.classList.remove('notselect')
            this.usageTypeRef.current.classList.remove('notselect')
        } else {
            this.usageTypeDropdownRef.current.classList.add('notselect')
            this.usageTypeRef.current.classList.add('notselect')
        }
    }

    /*** @description - This function is to handle the Assessment Usage-type change
     * @param usageType - the usage-type selected from the dropdown
     * @param e - event triggered 
    */
    handleAssessmentUsageTypeChange = (usageType, e) => {
        this.setState({
            activeAssessmentUsageType: usageType,
        });
        this.usageTypeDropdownRef.current.classList.add('notselect')
        this.usageTypeRef.current.classList.add('notselect')
        this.props.handleAssessmentBlur(usageType)
    
    }

    /*** @description - This function is to select the Assessment usage-type from dropdown*/
    selectAssessmentUsageType = () => {
         const { usageTypeListData } = this.props;
         let usageTypeDropdown = [], usageTypeValue = [];

         if (usageTypeListData && usageTypeListData.entityType == "assessment" && !(Object.keys(usageTypeListData.usageTypeList).length === 0 && usageTypeListData.usageTypeList.constructor === Object)) {
             usageTypeDropdown = Object.values(usageTypeListData.usageTypeList);
         }
         usageTypeValue = usageTypeDropdown && usageTypeDropdown.map((usageType, i) =>
             <li key={i} className="slate_assessment_metadata_dropdown_name" onClick={(e) => !hasReviewerRole() && this.handleAssessmentUsageTypeChange(usageType, e)}>{usageType}</li>
         )

        return usageTypeValue
    }

    /*** @description - This function is to render the Assessment Slate Element*/
    assessmentSlateContent = () => {
        if (document.getElementsByClassName("slate-tag-icon").length) {
        document.getElementsByClassName("slate-tag-icon")[0].classList.remove("disable");
        }
        let assessmentSlateJSX;
        let assessmentTypeValue;
        let changeTypeValue;
        let title = this.props.assessmentItemTitle? this.props.assessmentItemTitle:this.props.learningTemplateLabel;
        if(this.state.activeAssessmentType === LEARNING_APP_TYPE || this.state.activeAssessmentType === LEARNING_TEMPLATE){
            assessmentTypeValue="Learning App";
            changeTypeValue="Change Learning App"
        }else {
            assessmentTypeValue="Assessment";
            changeTypeValue="Change assessment";
        }
        if ((this.state.activeAssessmentType === FULL_ASSESSMENT_PUF || this.state.activeAssessmentType === LEARNOSITY) && this.state.showElmComponent === true) {
            return <RootElmSingleAssessment activeAssessmentType={this.state.activeAssessmentType} closeElmWindow = {()=>this.closeElmWindow()} addPufFunction = {this.addPufAssessment} activeUsageType = {this.state.activeAssessmentUsageType}/>
        }
        if ((this.state.activeAssessmentType === FULL_ASSESSMENT_CITE || this.state.activeAssessmentType === FULL_ASSESSMENT_TDX) && this.state.showCiteTdxComponent === true) {
            return <RootCiteTdxComponent activeAssessmentType={this.state.activeAssessmentType} openedFrom = {'slateAssessment'} closeWindowAssessment = {()=>this.closeWindowAssessment()} assessmentType = {this.state.activeAssessmentType} addCiteTdxFunction = {this.addCiteTdxAssessment} usageTypeMetadata = {this.state.activeAssessmentUsageType} parentPageNo={this.state.parentPageNo} isReset={this.state.isReset} resetPage={this.resetPage} AssessmentSearchTitle={this.AssessmentSearchTitle} searchTitle={this.state.searchTitle} filterUUID={this.state.filterUUID} />
        }
        if (this.props.getAssessmentData && this.props.getAssessmentDataPopup===false && this.state.changeLearningData === false) {
            assessmentSlateJSX = <div className="slate_fetch_canvas">
                <div className="slate_assessment_data_container">
                    <div className="slate_assessment_data_content">
                        <div className="slate_assessment_data_label">{assessmentTypeValue}</div>
                        <div className="slate_assessment_data_details">
                            <div className="slate_assessment_data_title">{title}</div>
                            <div className="slate_assessment_data_id">{'ID: ' + this.props.assessmentId}</div>
                            <div className="slate_assessment_data_id_lo" style={{display:"none"}}>{this.props.assessmentId}</div>
                            <div className="slate_assessment_data_format_lo" style={{display:"none"}}>{this.state.activeAssessmentType}</div>
                            <div className="slate_assessment_change_button" onClick={ !hasReviewerRole() && this.changeAssessment}>{changeTypeValue}</div>
                        </div>
                        <div className="clr"></div>
                    </div>
                </div>
                <div className="slate_assessment_metadata_container">
                    <div className="slate_assessment_metadata_type_selectlabel">Select usage type</div>
                    <div className="singleAssessment_Dropdown_activeDropdown notselect" ref={this.usageTypeRef} onClick={ !hasReviewerRole() && this.toggleUsageTypeDropdown} >
                        <span className="slate_assessment_metadata_dropdown_label" id ="AssessmentSlateUsageType">{this.state.activeAssessmentUsageType}</span>
                        <span className="slate_assessment_metadata_dropdown_image"></span>
                        <div className="clr"></div>
                    </div>
                    <div className="clr"></div>
                </div>
                {
                    <ul className="slate_assessment_metadata_type_dropdown_options notselect" ref={this.usageTypeDropdownRef}>
                        {this.selectAssessmentUsageType()}
                    </ul>
                }
                <div className="clr"></div>
            </div>
        }
        else if (this.state.changeLearningData && (this.state.activeAssessmentType === LEARNING_APP_TYPE || this.state.activeAssessmentType === LEARNING_TEMPLATE) && this.props.permissions && this.props.permissions.includes('quad_create_edit_ia')) {
            return (
                <div>
                    <LearningTool closePopUp={this.closePopUp} linkLearningApp={this.linkLearningApp} closelearningPopup={this.closelearningPopup}/>
                </div>
            )
        } else if ((this.props.getAssessmentData && this.props.getAssessmentDataPopup == true) || this.state.learningToolStatus) {
            assessmentSlateJSX = <div className="slate_popup_get_selection">
                <div className="slate_popup_get_image lazyload"></div>
                <div className="slate_popup_get_title">{"'" + title + "'"}</div>
                <div className="slate_popup_get_added">Successfully added</div>
                <div className="clr"></div>
            </div>
        } else {
            if (document.getElementsByClassName("slate-tag-icon").length) {
            document.getElementsByClassName("slate-tag-icon")[0].classList.add("disable");
            }
            assessmentSlateJSX = <div className="slate_initial_selection">
                <div className="slate_assessment_type_label">Please select an assessment type.</div>
                <div className="slate_assessment_type_dropdown activeDropdown notselect" ref={this.typeRef} onClick={this.toggleAssessmentTypeDropdown}>
                    <span className="slate_assessment_dropdown_label">{this.state.activeAssessmentType}</span>
                    <span className="slate_assessment_dropdown_image"></span>
                    <div className="clr"></div>
                </div>
                {
                    <ul className="slate_assessment_type_dropdown_options notselect" ref={this.typeDropdownRef}>
                        {this.selectAssessmentType()}
                    </ul>
                }
                {
                    this.state.activeAssessmentType != 'Select' ?
                        (<div className="slate_assessment_type_button" onClick={(e) => this.mainAddAssessment(e, this.state.activeAssessmentType)}>Add assessment</div>) : (<div className="slate_assessment_disabled_button" >Add assessment</div>)
                }
                <div className="clr"></div>
            </div>
        }
        return assessmentSlateJSX
    }
    render() {
        // const { type } = this.props;
        return (
            <div className="AssessmentSlateCanvas">
                {this.assessmentSlateContent()}
            </div>
        );
    }
}
AssessmentSlateData.displayName = "AssessmentSlateData"
const mapActionToProps = {
    setCurrentCiteTdx: setCurrentCiteTdx,
    assessmentSorting:assessmentSorting
}

export default connect(
    null,
    mapActionToProps
)(AssessmentSlateData);
