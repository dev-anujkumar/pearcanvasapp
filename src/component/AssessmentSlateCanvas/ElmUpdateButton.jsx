/**
* Component | ElmUpdateButton
* description | This is the Component for Update Button Field on Assessments for Elm
*/
/** ----- Import - Plugins ----- */
import React from 'react'
import { useEffect } from 'react';
/** ----- Import - Dependencies ----- */
import { approvedIcon } from '../../../src/images/ElementButtons/ElementButtons.jsx';
import './../../styles/AssessmentSlateCanvas/AssessmentSlateCanvas.css';
import { ELM_INT } from './AssessmentSlateConstants.js';

const ElmUpdateButton = (props) => {
    const { elmAssessment, updateElmVersion, buttonText, embeddedElmClass, elementType, status, isSubscribed, slateStatus } = props;

    useEffect(() => {
        if(elmAssessment.showUpdateStatus && status && !isSubscribed && slateStatus !== "approved"){
            updateElmVersion()
        }
    }, [])

    const setUpdateDiv = (assessment) => {
        let updateDiv;
        const { assessmentStatus, showUpdateStatus } = assessment
        const approveText = assessmentStatus == 'final' ? "Approved" : "Unapproved"
        const approveIconClass = assessmentStatus == 'final' ? "enable" : "disable"

        if (showUpdateStatus === true) {  
            updateDiv = <div className="eml-int-status-block">
                {(elementType === ELM_INT) && <div className={`elm-status-div ${embeddedElmClass}`}>
                        <p className="eml-int-status-text-tm">{approveText}</p>
                        <span className={"approved-button " + approveIconClass}>{approvedIcon}</span>
                    </div>
                }
                {!status && <div className={`elm-update-button ${embeddedElmClass}`} onClick={updateElmVersion}><b className='elm-update-button-text'>{buttonText}</b></div>}
                {(status && (isSubscribed || slateStatus === "approved")) && <div className={`elm-status-div ${embeddedElmClass}`}><span className={"approved-button " + approveIconClass}>{approvedIcon}</span><p className={"approved-button-text " + approveIconClass}>{approveText}</p></div>}
            </div>       
        } else {
            updateDiv = (elementType === ELM_INT) ? 
                <div className={`elm-status-div ${embeddedElmClass}`}>
                    <p className="eml-int-status-text-tm">{approveText}</p>
                    <span className={"approved-button " + approveIconClass}>{approvedIcon}</span>
                </div> :
                <div className={`elm-status-div ${embeddedElmClass}`}><span className={"approved-button " + approveIconClass}>{approvedIcon}</span><p className={"approved-button-text " + approveIconClass}>{approveText}</p></div>
        
        }
        return updateDiv
    }

    return (setUpdateDiv(elmAssessment));
}
export default ElmUpdateButton;