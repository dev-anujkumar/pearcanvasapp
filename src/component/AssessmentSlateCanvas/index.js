import AssessmentSlateCanvas from './AssessmentSlateCanvas.jsx';
export default AssessmentSlateCanvas;