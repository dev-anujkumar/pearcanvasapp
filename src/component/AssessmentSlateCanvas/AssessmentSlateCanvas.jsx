/**
* Module | AssessmentSlateCanvas
* description | This is the Root Component of Assessment Slate
*/
/** ----- Import - Plugins ----- */
import React, { Component } from 'react'
import { connect } from 'react-redux';
/** ----- Import - Components ----- */
import TinyMceEditor from "./../tinyMceEditor";
import AssessmentSlateData from './AssessmentSlateData.jsx';
/** ----- Import - Dependencies ----- */
import { specialCharacterDecode } from './assessmentCiteTdx/Actions/CiteTdxActions';
import { showTocBlocker, disableHeader, hideTocBlocker, hideToc, ShowCanvasLoader } from '../../js/toggleLoader';
import { setAssessmentFormat, hasAssessmentID, setAssessmentElement, setAssessmentUsageType } from './AssessmentActions/assessmentUtility.js';
/*** @description - AssessmentSlateCanvas is a class based component that acts as a container for Assessment Slate */
class AssessmentSlateCanvas extends Component {
    constructor(props) {
        super(props);
        this.state = {
            assessmentFormat: props.model && setAssessmentFormat(props.model),
            getAssessmentData: props.model && hasAssessmentID(props.model),
            assessmentSlateObj: props.model && setAssessmentElement(props.model),
            getAssessmentDataPopup: false
        }
    }

    static getDerivedStateFromProps(nextProps) {
        if ('model' in nextProps && 'elementdata' in nextProps.model && 'assessmentformat' in nextProps.model.elementdata) {
            return {
                assessmentFormat: nextProps.model && setAssessmentFormat(nextProps.model),
                getAssessmentData: nextProps.model && hasAssessmentID(nextProps.model),
                assessmentSlateObj: nextProps.model && setAssessmentElement(nextProps.model)
            }
        }
    }

    /*** 
    * @description - This is the function to add Elm/Learnosity Assessments to Assessment Slate 
    * @param pufObj - The object contains data about Elm/Learnosity Assessment 
    */
    addPufAssessment = (pufObj, activeAssessmentType, updateType, cb) => {
        showTocBlocker();
        disableHeader(true);
        this.updateAssessment(pufObj.id, "", pufObj.title, activeAssessmentType, pufObj.usagetype, updateType);
        if (cb) {
            cb();
        }
    }

    /*** 
    * @description - This is the function to add CITE/TDX Assessments to Assessment Slate 
    * @param citeTdxObjs - The object contains data about CITE/TDX Assessment 
    * @param activeAssessmentType currwnt assessment format (CITE/TDX)
    */
    addCiteTdxAssessment = (citeTdxObj, activeAssessmentType) => {
        showTocBlocker();
        disableHeader(true);
        this.updateAssessment(citeTdxObj.id, "", specialCharacterDecode(citeTdxObj.title), activeAssessmentType, citeTdxObj.usageType, 'insert');
    }

    /*** @description - This function is to update state variables based on the parameters
       * @param id - assessment-id of the assessment
       * @param itemID - assessment-item-id of the assessment
       * @param title - assessment-title of the assessment
       * @param format - assessment Format of the assessment
       * @param usageType - usageType of the assessment
       * @param change - type of change - insert/update
    */
    updateAssessment = (id, itemID, title, format, usageType, change, learningsystem, templateid, templatetype) => {
        let dataToSend = {
            id: id,
            title: title,
            itemID: itemID,
            format: format,
            usageType: usageType,
            templatelabel: title,
            templateid: templateid,
            templatetype: templatetype,
            learningsystem: learningsystem
        }
        this.setState({
            assessmentSlateObj: {
                assessmentId: id,
                itemId: itemID,
                title: title,
            },
            getAssessmentData: true,
        }, () => {
            this.handleAssessmentBlur(dataToSend);
        })
        if (change === 'insert') {
            this.setState({
                getAssessmentDataPopup: true
            }, () => {
                setTimeout(() => {
                    this.setState({
                        getAssessmentDataPopup: false
                    })
                }, 3000)
                ShowCanvasLoader(false);
            })
        }
        else {
            this.setState({
                getAssessmentData: false
            })
        }
        disableHeader(false);
        hideTocBlocker(false);
    }

    /*** @description - This function is to handle Focus on the Assessment element on click*/
    handleAssessmentFocus = () => {
        this.props.handleFocus();
    }

    /*** @description - This function is to handle Blur on the Assessment element on blur*/
    handleAssessmentBlur = (assessmentData, cb) => {
        this.props.handleBlur(assessmentData);
        if (assessmentData.calledFrom == 'updateAssessmentFormat') {
            this.setState({
                getAssessmentData: false,
                getAssessmentDataPopup: false
            })
            if (cb) {
                cb();
            }
        }
    }

    render() {
        const { type, model, permissions, isLOExist, showBlocker, elementId } = this.props;
        const { getAssessmentDataPopup, getAssessmentData, assessmentSlateObj } = this.state;
        let handleCanvasBlocker = {
            showTocBlocker: showTocBlocker,
            disableHeader: disableHeader,
            hideTocBlocker: hideTocBlocker,
            hideToc: hideToc,
            showBlocker: showBlocker,
            ShowLoader: ShowCanvasLoader
        }
        return (
            <div className="AssessmentSlateMenu" onClick={this.handleAssessmentFocus}>
                <AssessmentSlateData
                    type={type}
                    model={model}
                    isLOExist={isLOExist}
                    permissions={permissions}
                    getAssessmentData={getAssessmentData}
                    assessmentSlateObj={assessmentSlateObj}
                    getAssessmentDataPopup={getAssessmentDataPopup}
                    updateAssessment={this.updateAssessment}
                    addPufAssessment={this.addPufAssessment}
                    setAssessmentFormat={setAssessmentFormat}
                    handleCanvasBlocker={handleCanvasBlocker}
                    handleAssessmentBlur={this.handleAssessmentBlur}
                    addCiteTdxAssessment={this.addCiteTdxAssessment}
                    setAssessmentUsageType={setAssessmentUsageType}
                    elementId={elementId}
                />
                <TinyMceEditor
                    slateLockInfo={this.props.slateLockInfo}
                    handleBlur={this.props.handleBlur}
                    model={this.props.model}
                    handleEditorFocus={this.props.handleFocus}
                    className="addLOdata"
                    permissions={this.props.permissions}
                    element={this.props.model}
                />
            </div>
        );
    }
}

AssessmentSlateCanvas.displayName = "AssessmentSlateCanvas"

const mapStateToProps = (state) => {
    return {
        permissions: state.appStore.permissions
    }
}

export default connect(mapStateToProps, {})(AssessmentSlateCanvas)
