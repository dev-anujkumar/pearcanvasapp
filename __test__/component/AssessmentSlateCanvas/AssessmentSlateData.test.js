import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { AssessmentSlateData } from '../../../src/component/AssessmentSlateCanvas/AssessmentSlateData';
import {assessmentSlateDefault,assessmentSlateWithData} from "./../../../fixtures/AssessmentSlateCanvasTestingData";


describe('Testing Assessment Slate Data component', () => {
    const div = document.createElement('div');
    const assessmentSlate = mount( <AssessmentSlateData   />)
    let assessmentSlateDataInstance = assessmentSlate.find('AssessmentSlateData').instance();
    it('render Assessment Slate Data component ', () => {
        
            ReactDOM.render(<AssessmentSlateData />, div);
            ReactDOM.unmountComponentAtNode(div);
    })
    
    it('onClick Assessment Type Event', () => {
        assessmentSlateDataInstance = assessmentSlate.find('AssessmentSlateData').instance();
        assessmentSlateDataInstance.setState({
        activeAssessmentType: 'Select'
    });

    let activeAssessmentType = {
        activeAssessmentType: {
            getAttribute: function(dataValue) {
                return 'Full Assessment CITE';
            }
        }
    }
       
        assessmentSlate.find('div.slate_assessment_type_dropdown.activeDropdown').simulate('click');
        assessmentSlate.find('ul.slate_assessment_type_dropdown_options>li:first-child').simulate('click');
   });   
it('onClick UsageType Event', () => {
    let props={
        getAssessmentDataPopup :false,
        assessmentId:"urn:pearson:work:4da32e71-a6b5-4daa-84ed-fb72d6b0aa74",
        assessmentItemTitle:"1.1 Homework",
        model:assessmentSlateWithData,
        getAssessmentData:true,
        
    }
    const component = mount(<AssessmentSlateData {...props} />);
    assessmentSlateDataInstance.setState({
        activeAssessmentType:"Full Assessment CITE",
        activeAssessmentUsageType: 'Quiz'
    });
    assessmentSlateDataInstance.forceUpdate();
    
    assessmentSlate.find('div.singleAssessment_Dropdown_activeDropdown').simulate('click');
    assessmentSlate.find('ul.slate_assessment_metadata_type_dropdown_options>li:first-child').simulate('click');


        
});
    // it('Select PUF assessment', () => {
    //     const component = mount(<AssessmentSlateData {...props} />);
    //     assessmentSlateDataInstance.setState({
    //         activeAssessmentType: 'Full Assessment PUF',
    //         showElmComponent: true,
    //     });
    //     assessmentSlateDataInstance.forceUpdate();

    // });
    it('Render succcessfully addedd screen', () => {
        let props = {
            getAssessmentDataPopup: true,
            assessmentId: "urn:pearson:work:4da32e71-a6b5-4daa-84ed-fb72d6b0aa74",
            assessmentItemTitle: "1.1 Homework",
            model: assessmentSlateWithData,
            getAssessmentData: true,

        }
        const component = mount(<AssessmentSlateData {...props} />);
        assessmentSlateDataInstance.setState({
            activeAssessmentType: 'Full Assessment PUF',
            showElmComponent: true,
        });
        assessmentSlateDataInstance.forceUpdate();
    });
    it('Active assessment type handling', () => {
        let props = {
            getAssessmentData : true,
            getAssessmentDataPopup : false,
        }
        const component = mount(<AssessmentSlateData {...props} />);
        assessmentSlateDataInstance.setState({
            activeAssessmentType : "Learning App Type"   
        });

        assessmentSlateDataInstance.forceUpdate();
    });
})