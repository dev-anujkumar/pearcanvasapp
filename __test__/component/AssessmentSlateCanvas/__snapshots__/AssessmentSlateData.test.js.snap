// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Assessment Slate Data component render Assessment Slate Data component  1`] = `
<AssessmentSlateData>
  <div
    className="AssessmentSlateCanvas"
  >
    <div
      className="slate_initial_selection"
    >
      <div
        className="slate_assessment_type_label"
      >
        Please select an assessment type.
      </div>
      <div
        className="slate_assessment_type_dropdown activeDropdown"
        onClick={[Function]}
      >
        <span
          className="slate_assessment_dropdown_label"
        >
          Select
        </span>
        <span
          className="slate_assessment_dropdown_image"
        />
        <div
          className="clr"
        />
      </div>
      <ul
        className="slate_assessment_type_dropdown_options notselect"
      >
        <li
          className="slate_assessment_dropdown_name"
          key="0"
          onClick={[Function]}
        >
          Full Assessment CITE
        </li>
        <li
          className="slate_assessment_dropdown_name"
          key="1"
          onClick={[Function]}
        >
          Full Assessment TDX
        </li>
        <li
          className="slate_assessment_dropdown_name"
          key="2"
          onClick={[Function]}
        >
          Full Assessment PUF
        </li>
        <li
          className="slate_assessment_dropdown_name"
          key="3"
          onClick={[Function]}
        >
          Learning App Type
        </li>
      </ul>
      <div
        className="slate_assessment_disabled_button"
      >
        Add assessment
      </div>
      <div
        className="clr"
      />
    </div>
  </div>
</AssessmentSlateData>
`;
